src/arkode/arkode_io.c

```diff
@@ -34,6 +35,347 @@
   ARKODE optional input functions
   ===============================================================*/
 
+/*---------------------------------------------------------------
+  ARKodeSetFromCommandLine:
+
+  Parses the command line to control scalar-valued ARKODE options.
+
+  (this leverages a few typedefs and static utility routines)
+  ---------------------------------------------------------------*/
+typedef int (*arkIntSetFn)(void*,int);
+typedef int (*arkLongSetFn)(void*,long int);
+typedef int (*arkRealSetFn)(void*,sunrealtype);
+typedef int (*arkBoolSetFn)(void*);
+static int CheckAndSetIntArg(ARKodeMem ark_mem, int *i, char *argv[],
+                             const char* argtest, arkIntSetFn fname,
+                             sunbooleantype *arg_used)
+{
+  *arg_used = SUNFALSE;
+  if (strcmp(argv[*i], argtest) == 0)
+  {
+    int iarg = atoi(argv[++(*i)]);
+    if (fname((void*) ark_mem, iarg) != ARK_SUCCESS)
+    {
+      arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                      "error setting command-line argument: %s %s",
+                      argv[(*i)-1], argv[*i]);
+      return ARK_ILL_INPUT;
+    }
+    *arg_used = SUNTRUE;
+  }
+  return ARK_SUCCESS;
+}
+
+static int CheckAndSetLongArg(ARKodeMem ark_mem, int *i, char *argv[],
+                              const char* argtest, arkLongSetFn fname,
+                              sunbooleantype *arg_used)
+{
+  *arg_used = SUNFALSE;
+  if (strcmp(argv[*i], argtest) == 0)
+  {
+    long int iarg = atol(argv[++(*i)]);
+    if (fname((void*) ark_mem, iarg) != ARK_SUCCESS)
+    {
+      arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                      "error setting command-line argument: %s %s",
+                      argv[(*i)-1], argv[*i]);
+      return ARK_ILL_INPUT;
+    }
+    *arg_used = SUNTRUE;
+  }
+  return ARK_SUCCESS;
+}
+
+static int CheckAndSetRealArg(ARKodeMem ark_mem, int *i, char *argv[],
+                              const char* argtest, arkRealSetFn fname,
+                              sunbooleantype *arg_used)
+{
+  *arg_used = SUNFALSE;
+  if (strcmp(argv[*i], argtest) == 0)
+  {
+    sunrealtype rarg = atof(argv[++(*i)]);
+    if (fname((void*) ark_mem, rarg) != ARK_SUCCESS)
+    {
+      arkProcessError(ark_mem, ARK_ILL_INPUT, __LINE__, __func__, __FILE__,
+                      "error setting command-line argument: %s %s",
+                      argv[(*i)-1], argv[*i]);
+      return ARK_ILL_INPUT;
+    }
+    *arg_used = SUNTRUE;
+  }
+  return ARK_SUCCESS;
+}
+
+static int CheckAndSetBoolArg(ARKodeMem ark_mem, int *i, char *argv[],
+                              const char* argtest, arkBoolSetFn fname,
+                              sunbooleantype *arg_used)
+{
+  *arg_used = SUNFALSE;
+  if (strcmp(argv[*i], argtest) == 0)
```

Comment:
So booleans are specified with just the flag? I think it would be preferable to have flag-value pair command line syntax for boolean options. Otherwise, a user will need to know the default and whether to override it.

Reply:
I think it would be clearer to refer to these a "aciton" functions rather than boolean functions as their only input is the integrator memory structure e.g., `ARKodeSetNonlinear` and `ARKodeClearStopTime`. The functions that take a boolean input are group with the "int" functions.